### PR TITLE
BTM-587 & BTM-596 Extend Frontend prototype lambda

### PR DESCRIPTION
**Pull request by Si3rr4wow**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -75,7 +90,11 @@
   "dependencies": {
     "@aws-lambda-powertools/logger": "^1.6.0",
     "@aws-sdk/util-utf8-node": "^3.259.0",
+    "@vendia/serverless-express": "^4.10.1",
+    "express": "^4.18.2",
+    "govuk-frontend": "^4.6.0",
     "mailparser": "^3.6.4",
-    "redux-thunk": "^2.4.2"
+    "nunjucks": "^3.2.4",
+    "style-loader": "^3.3.3"
```
[the remark this answers is not in the document]
Comment (Si3rr4wow) [reply package.json:101]: good shout, I'll move it upp

File: src/frontend/app.ts [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+app.get("/invoices", (_, response) => {
+  response.render("invoices.njk");
+});
+app.get("/invoices/:id", (_, response) => {
+  console.log(_.params);
+  response.render("invoice.njk");
+});
+
+app.use("/images", express.static(dirname));
+app.use("/scripts", express.static(dirname));
```
[the remark this answers is not in the document]
Comment (Si3rr4wow) [reply src/frontend/app.ts:48]: Definitely not what I intended. Good comment, also made me realise the images and fonts are in an orphaned assets dir

File: src/frontend/package-lock.json
[the remark this answers is not in the document]
Comment (Si3rr4wow) [reply src/frontend/package-lock.json:1]: Yeah idk how this got in there, removed

File: src/frontend/app.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (Si3rr4wow) [reply src/frontend/app.ts]: reshuffled some bits and bobs, all working as expected now

File: src/frontend/views/contracts.njk [hunk 3]
```diff
@@ -0,0 +1,11 @@
+{% extends "common/layout/base.njk" %}
+
+{% block content %}
+  <h1 class="govuk-heading-l">Digital Identity Billing Dashboard</h1>
+  <p class="govuk-body">Welcome to the contracts page!</p>
+  <ul>
+  {% for contract in contracts %}
+  <li>{{ contract.name }}</li>
+{% endfor %}
+</ul>
```
[the remark this answers is not in the document]
Comment (Si3rr4wow) [reply src/frontend/views/contracts.njk]: Done and added djlint to solve in future

File: src/loader.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+  const stat = fs.statSync(path, {
+    throwIfNoEntry: false,
+  })
+  if (stat?.isDirectory() && !shouldEmpty) {
+    return
+  } else if (stat?.isDirectory() && shouldEmpty) {
+    fs.rmSync(path, { recursive: true })
+  }
+  fs.mkdirSync(path, { recursive: true })
+}
```
[the remark this answers is not in the document]
Comment (Si3rr4wow) [reply src/loader.js]: I've removed this file now, it was just to help with debugging the webpack loaders

File: webpack.config.cjs
[the remark this answers is not in the document]
Comment (Si3rr4wow) [reply webpack.config.cjs:1]: For various just-get-it-working related reasons, I think we can switch it back now though. Lemme try some stuff out

File: assets/fonts/bold-affa96571d-v2.woff
[the remark this answers is not in the document]
Comment (Si3rr4wow) [reply assets/fonts/bold-affa96571d-v2.woff:1]: They come from the gov uk node module, followed instructions here https://frontend.design-system.service.gov.uk/get-started/#4-get-the-font-and-images-working

I agree no es bueno. Lemme try something out

[the remark this answers is not in the document]
Comment (Si3rr4wow) [reply assets/fonts/bold-affa96571d-v2.woff:1]: An update on this. Now that I've moved these into assets under src the build is broken. Turns out the scss expects them to be in a certain place. I'm tryna get the webpack copy plugin to move them at build time but I can't figure how to orchestrate that copy happening before the sass loader kicks in and screams

File: template-source.yaml [hunk 5]
```diff
@@ -118,4 +118,8 @@ Resources: !YAMLInclude ./cloudformation/global.yaml,
   ./cloudformation/custom-athena-view-resource.yaml#NO_LOCAL,
   ./cloudformation/invoice-athena.yaml#NO_LOCAL,
   ./cloudformation/email-storage.yaml,
-  ./cloudformation/dashboard-data-extraction.yaml#NO_LOCAL
+  ./cloudformation/dashboard-data-extraction.yaml#NO_LOCAL,
+  ./cloudformation/frontend.yaml
+# Outputs:
+#   FrontEndUrl:
+#     Value: !Sub https://${FrontEndApiSubdomain}.execute-api.${AWS::Region}.amazonaws.com
```
[the remark this answers is not in the document]
Comment (MelAtGDS) [reply template-source.yaml:125]: unsure what it will be needed for. removed for now.

**Review by markschnitziusgovuk — APPROVED** [review]
This LGTM but probably needs more eyes.  I'll approve it though so you can merge if you want.

**Pull request merged**